### PR TITLE
Fork CUDA and ROCm guides into separate pages.

### LINKED ISSUE #15115: Split CUDA/ROCm website page into two
As more work is being done to modernize the CUDA and ROCm stacks in IREE, https://www.iree.dev/guides/deployment-configurations/gpu-cuda-rocm/ can be split into two separate pages. The pages may still look very similar, but I think they're ready to stand on their own, especially now that we have a "Guides" top level website category helping keep the nav organized.

---

A tiny detail I've been thinking about is what icons to use for the pages

image

I'd like to use NVIDIA and AMD brand icons, but the AMD icon from https://simpleicons.org/ is difficult to read at this scale:
image

(AMD brand guidelines say to use the full logo with the text and not just the mark: https://www.amd.com/system/files/2017-06/AMD_BrandGuidelines_Original_34751.pdf, but this usage is a square icon, so I could see a case for the mark/favicon instead of the full logo...)

### PR DESCRIPTION
**Pull request by ScottTodd**
Fixes https://github.com/openxla/iree/issues/15115

These pages are very similar, but both stacks have undergone enough development recently that they can now stand on their own. This PR primarily forks the pages, but it also does some minor cleanup (fixing dead links, adjusting formatting).

|  | |
|--------|--------|
| Current page | https://www.iree.dev/guides/deployment-configurations/gpu-cuda-rocm/ |
| Preview of this PR - CUDA | https://scotttodd.github.io/iree/guides/deployment-configurations/gpu-cuda | 
| Preview of this PR - ROCm | https://scotttodd.github.io/iree/guides/deployment-configurations/gpu-rocm |

### CODE REVIEW

File: docs/website/docs/guides/deployment-configurations/gpu-cuda.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+and convert it using IREE's
+[TensorFlow importer](../ml-frameworks/tensorflow.md). Then run one of the
+following commands to compile:
+
+```shell hl_lines="2-3"
+iree-compile \
+    --iree-hal-target-backends=cuda \
+    --iree-hal-cuda-llvm-target-arch=<...> \
+    mobilenet_iree_input.mlir -o mobilenet_cuda.vmfb
+```
```
Comment (ScottTodd) [inline docs/website/docs/guides/deployment-configurations/gpu-cuda.md:112]: These TF sections should be replaced with something more generic. Future work...

**Comment by raikonenfnu** [comment]
> Fixes #15115
> 
> These pages are very similar, but both stacks have undergone enough development recently that they can now stand on their own. This PR primarily forks the pages, but it also does some minor cleanup (fixing dead links, adjusting formatting).
> 
> Current page	https://www.iree.dev/guides/deployment-configurations/gpu-cuda-rocm/
> Preview of this PR - CUDA	https://scotttodd.github.io/iree/guides/deployment-configurations/gpu-cuda
> Preview of this PR - ROCm	https://scotttodd.github.io/iree/guides/deployment-configurations/gpu-rocm

Tbh, both pages still looks quite similar to me for now haha, I do think it's good to start splitting them up for better legibility and UI/UX. :)

Just curious, are there any other reasons you started splitting the page up?

**Comment by ScottTodd** [comment]
> > Fixes #15115
> > These pages are very similar, but both stacks have undergone enough development recently that they can now stand on their own. This PR primarily forks the pages, but it also does some minor cleanup (fixing dead links, adjusting formatting).
> > Current page	https://www.iree.dev/guides/deployment-configurations/gpu-cuda-rocm/
> > Preview of this PR - CUDA	https://scotttodd.github.io/iree/guides/deployment-configurations/gpu-cuda
> > Preview of this PR - ROCm	https://scotttodd.github.io/iree/guides/deployment-configurations/gpu-rocm
> 
> Tbh, both pages still looks quite similar to me for now haha, I do think it's good to start splitting them up for better legibility and UI/UX. :)
> 
> Just curious, are there any other reasons you started splitting the page up?

Yeah, I could go either way here. On one end, we could fold all of the GPU pages (Vulkan, CUDA, ROCm, Metal) into a single page, and on the other (this PR), we could split them into separate pages. We're currently somewhere in the middle, since CUDA and ROCm were both at very similar stages of (early) development.

The main reason I want separate pages is for clear messaging to users that each backend can stand on its own and can provide differentiating features.

* Vulkan: interop with graphics APIs, portability to platforms like Android
* CUDA: multi-GPU setups (collectives), plugins/comparisons/integrations with other NVIDIA libraries
* ROCm: AMD GPU support with specialization not possible via Vulkan
* Metal: direct support on Apple platforms, without going through compatibility layers (MoltenVK)

The implementations are (by design) quite similar in IREE, and some users may appreciate side by side comparisons, but I expect that many users will already know what hardware/APIs they are using and would rather see documentation specialized for that.

**Comment by raikonenfnu** [comment]
SGTM!

**Review by raikonenfnu — APPROVED** [review]

**Pull request merged**